sys/net/gnrc/link_layer/lorawan/gnrc_lorawan_mlme.c

```diff
@@ -237,43 +262,48 @@ void gnrc_lorawan_mlme_request(gnrc_lorawan_t *mac,
                                mlme_confirm_t *mlme_confirm)
 {
     switch (mlme_request->type) {
-        case MLME_JOIN:
-            if (mac->mlme.activation != MLME_ACTIVATION_NONE) {
-                mlme_confirm->status = -EINVAL;
-                break;
-            }
-            if (!gnrc_lorawan_mac_acquire(mac)) {
-                mlme_confirm->status = -EBUSY;
-                break;
-            }
-
-            if (mac->mlme.backoff_budget < 0) {
-                mlme_confirm->status = -EDQUOT;
-                break;
-            }
-            memcpy(mac->appskey, mlme_request->join.appkey, LORAMAC_APPKEY_LEN);
-            mlme_confirm->status = gnrc_lorawan_send_join_request(mac,
-                                                                  mlme_request->join.deveui,
-                                                                  mlme_request->join.appeui, mlme_request->join.appkey,
-                                                                  mlme_request->join.dr);
-            break;
-        case MLME_LINK_CHECK:
-            mac->mlme.pending_mlme_opts |=
-                GNRC_LORAWAN_MLME_OPTS_LINK_CHECK_REQ;
-            mlme_confirm->status = GNRC_LORAWAN_REQ_STATUS_DEFERRED;
-            break;
-        case MLME_SET:
-            _mlme_set(mac, mlme_request, mlme_confirm);
-            break;
-        case MLME_GET:
-            _mlme_get(mac, mlme_request, mlme_confirm);
+    case MLME_JOIN:
+        if (mac->mlme.activation != MLME_ACTIVATION_NONE) {
+            mlme_confirm->status = -EINVAL;
             break;
-        case MLME_RESET:
-            gnrc_lorawan_reset(mac);
-            mlme_confirm->status = GNRC_LORAWAN_REQ_STATUS_SUCCESS;
+        }
+        if (!gnrc_lorawan_mac_acquire(mac)) {
+            mlme_confirm->status = -EBUSY;
             break;
-        default:
+        }
+
+        if (mac->mlme.backoff_budget < 0) {
+            mlme_confirm->status = -EDQUOT;
             break;
+        }
+
+#if IS_USED(MODULE_GNRC_LORAWAN_1_1)
+        memcpy(mac->appskey, mlme_request->join.appkey, LORAMAC_APPKEY_LEN);
+#endif
```

Comment:
please use if(IS_USED) here. Although there's an overhead of adding a getter for join->appkey, we catch for free compilation errors.
Also, it doesn't seem to be the first part where the code requires these struct members. So, it's IMO definitely worth it to use getters here.